data/scripts/bin/update-hosts.sh.template

```diff
@@ -49,13 +36,39 @@ host_env_base64=$(base64 -w 0 /etc/assisted/rendezvous-host.env)
 placeholder_base64=$(echo -n '{{.RendezvousHostEnvPlaceholder}}' | base64)
 ignition="${ignition//$placeholder_base64/$host_env_base64}"
 
-# Update installer-args and ignition for each host
-if [[ -n ${host_ids} ]]; then
+# Waiting for the cluster-id to be available
+until [[ -n ${cluster_id} ]]; do
+    echo "Querying assisted-service for cluster-id..."
+    cluster_id=$(curl_assisted_service "/clusters" GET | jq -r '.[0].id')
+    sleep 1
+done
+echo "Fetched cluster-id: $cluster_id"
+
+# Updating hosts can be done before starting the installation
+until [[ $cluster_status == "preparing-for-installation" ]]; do
+    cluster_status=$(curl_assisted_service "/clusters/${cluster_id}" | jq -r .status)
+
+    # Update ignition for each host
+    host_ids=$(curl_assisted_service "/infra-envs/${INFRA_ENV_ID}/hosts" | jq -r .[].id)
+    if [[ -z ${host_ids} ]]; then
+        sleep 2
+        continue
+    fi
+
     for id in ${host_ids}; do
+        if [[ ${hosts[$id]} == "true" ]]; then
+            # Host is already updated
+            continue
+        fi
+
         # Update host's ignition (used when booting from the installation disk after bootstrap)
-        curl_assisted_service "${BASE_URL}/infra-envs/${INFRA_ENV_ID}/hosts/${id}/ignition" \
+        curl_assisted_service "/infra-envs/${INFRA_ENV_ID}/hosts/${id}/ignition" \
             PATCH -d '{"config": '"${ignition}"'}'
-
-        echo "Updated installer-args and ignition of host with id: ${id}"
+        hosts[$id]=true
+        echo "Updated ignition of host: ${id}"
     done
-fi
+
+    sleep 2
```

Comment:
```suggestion
    sleep 1
```
Just to reduce the waiting time before discovering a new host needing to be patched

Reply:
Sure, changed.